datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -463,6 +487,38 @@ public String reduce(String query, Map<String, String> responses) {
     return null;
   }
 
+  private String reduceTopicPartitionStatsResponses(Map<String, String> responses) {
+    Map<String, Map<String, Set<Integer>>> result = new HashMap<>();
```

Comment:
Is the endpoint invoked for a particular datastream? If so, what are the cases where we would expect more than one consumer group? 

If the response is supposed to include data for multiple unrelated datastreams, I think the response should include datastream name as well?

Reply:
The end point is not invoked for a particular datastream. It just aggregates the consumer groups across all brooklin instances, datastreams.  

If the response is supposed to include data for multiple unrelated datastreams, I think the response should include datastream name as well?
I believe the intent here is to get enough info to query Kafka or lag monitor.  Since datastream is an internal brooklin concept it will not be reported.

Reply:
I believe datastream name will be useful, as that's a fundamental unit of operation for brooklin (for example, if one needs to restart a datstream to recover from lags).
Admittedly there are other ways to map datastream to consumer group and some type of group IDs even have datastream names in them, but that can mean going through all the datastreams in a cluster. 
We can discuss this offline and add the datastream in a separate PR if needed.